components/RichTextEditor.js

```diff
@@ -286,6 +287,61 @@ export default class RichTextEditor extends React.Component {
     }
   }
 
+  /** A helper used by autolink to find where the url actually ends
+   * Credits:
+   * https://github.com/github/cmark-gfm/blob/36c1553d2a1f04dc1628e76b18490edeff78b8d0/extensions/autolink.c#L37
+   * https://github.com/vmg/redcarpet/blob/92a7b3ae2241b862e9bf45e0af3cf53ebdfb0afb/ext/redcarpet/autolink.c#L58
+   */
+  autolinkDelimiter = url => {
+    let linkEnd = url.length;
+
+    while (linkEnd > 0) {
+      const cclose = url[linkEnd - 1];
+
+      let copen;
+      switch (cclose) {
+        case '"':
+          copen = '"';
+          break;
+        case "'":
+          copen = "'";
+          break;
+        case ')':
+          copen = '(';
+          break;
+        case ']':
+          copen = '[';
+          break;
+        case '}':
+          copen = '{';
+          break;
+      }
```

Comment:
**[:ice_cream: Details & personal preferences]** I think that this would be easier to read with a `if` `else if`.

---

<details>
<summary>More details about this feedback</summary>
<br/>

This feedback categorization is inspired by [Netlify's Feedback Ladders system](https://www.netlify.com/blog/2020/03/05/feedback-ladders-how-we-encode-code-reviews-at-netlify). It includes 5 levels of feedback:

:rotating_light: **Critical blocker**
This feedback blocks all related work, and requires immediate action. 

:no_entry_sign: **Blocker**
This feedback should block the work from moving forward or being approved, but it does not necessarily require immediate action from the team.

:hammer_and_wrench: **Important**
This feedback should not block the work from moving forward or being approved, but it does require future action from the team. This generally means the feedback is not required for the MVP or initial draft of this work, but should be addressed at some point and we should ideally create an issue to track it.

:sparkles: **Enhancement**
This feedback should not block the work from moving forward or being approved. The project lead or owner should consider the feedback and implement or open an issue it if they agree with it.

:ice_cream: **Details & personal preferences**
This feedback should not block the work from moving forward or being approved. The project lead or owner should consider the feedback and implement or open an issue it if they deem it useful. This is also the category for everything that relates to personal preferences or code style.

</details>

Reply:
Because the code is coming from somewhere else, I'm ok keeping it structured like that.

Reply:
Not a blocker for me neither, but I think it's better to clean the code that we copy paste to make it easier to read and understand. I had a hard time understanding in depth what this function is doing.

Reply:
I can work on improving readability of this